### PR TITLE
FFT photon ops: Diffraction

### PR DESCRIPTION
**Pull request by g-braeunlich**
Closes: #299

### CODE REVIEW

File: imsim/stamp.py [hunk 1]
```diff
@@ -480,6 +480,11 @@ def fix_seds(prof):
             # can blow up the required FFT size.  We'll guard for that below with
             # a try block, but we can minimize how often this happens by making sure
             # the offset is close to 0,0.
```
Comment (rmjarvis) [inline imsim/stamp.py:482]: This comment is now in the wrong place.  It refers to the below test re offset.  But I think we agreed a while back that that block is broken and we should just remove it.  You can probably go ahead and do that on this PR.

File: imsim/photon_ops.py [hunk 2]
```diff
@@ -200,7 +200,7 @@ def applyTo(self, photon_array, local_wcs=None, rng=None):
             geometry=LSST_SPIDER_GEOMETRY,
             distribution=self.diffraction_rng(rng),
         )
-        photon_array.x, photon_array.y = xy_to_v.inverse(v)
+        photon_array.x[()], photon_array.y[()] = xy_to_v.inverse(v)
```
Comment (rmjarvis) [inline imsim/photon_ops.py:203]: Why this change?

File: imsim/stamp.py [hunk 3]
```diff
@@ -494,7 +499,11 @@ def fix_seds(prof):
                                method='fft',
                                offset=fft_offset,
                                image=fft_image,
-                               wcs=wcs)
+                               wcs=wcs,
+                               photon_ops=photon_ops,
+                               sensor=galsim.Sensor(),
```
Comment (rmjarvis) [inline imsim/stamp.py:504]: This won't be necessary once GalSim 2.4.6 is released.  I fixed the test that had been triggering the assert failure.

File: config/imsim-config.yaml [hunk 4]
```diff
@@ -198,6 +198,13 @@ stamp:
     world_pos:
         type: InstCatWorldPos
 
+    fft_photon_ops:
+        -
+            type: lsst_diffraction
+            telescope: LSST
+            band: { type: OpsimMeta, field: band }
+            latitude: -30.24463
```
Comment (rmjarvis) [inline config/imsim-config.yaml:206]: As mentioned on the Galsim PR, this photon op requires pupil sampling, so you need to include a pupil sampler op here too.  Probably also a time sampler.

**Review by rmjarvis — CHANGES_REQUESTED** [review]

File: imsim/photon_ops.py (hunk 2 above)
Comment (g-braeunlich) [reply imsim/photon_ops.py:203]: Indeed, I think, here we dont gain anything by this change.
Before: rhs is a new numpy.ndarray which then will replace lhs which should be garbage collected.
After: rhs is again a new numpy.ndarray which then will be copied to lhs and garbage collected afterwards.

Will remove it again

Comment (rmjarvis) [reply imsim/photon_ops.py:203]: Actually, the setter for x and y already write into the existing array.  So in both cases, the temporary returned by xy_to_v.inverse is the one that will be garbage collected.

Comment (g-braeunlich) [reply imsim/photon_ops.py]: OK. I removed this commit.

File: imsim/stamp.py (hunk 3 above)
Comment (rmjarvis) [reply imsim/stamp.py:494]: One other thing here.  The photon sampling of the FFT image uses an rng, so you should also pass `rng=self.rng` here.

**Comment by rmjarvis** [comment]
I would have liked to see a test that just ran the code in fft mode using the config without all this mock stuff.  I seriously cannot tell what the test is even testing, since practically everything is mocked up.  But that's probably just a style difference -- I prefer things that are closer to integration tests, whereas this is closer to what TDD people would call a pure unit test.  Since Chris is in a hurry to merge this, let's just call this fine.  It probably does test the actual bits that you changed in this PR.
Anyway, I think you need to mark it "Ready for review" though, since it's still a work in progress, so I can't officially sign off on it yet.

**Comment by g-braeunlich** [comment]
@rmjarvis Thx for the review! I now marked this PR as ready

**Review by rmjarvis — APPROVED** [review]
